### PR TITLE
Add webpack

### PR DESCRIPTION
**Pull request by Oliphaunte**
Will add Sass stuff in a little bit

### CODE REVIEW

**Comment by Rulox** [comment]
We have to make more changes.

* Delete browsersync and gulpfile
* Use webpack server
* Use hotloader for components

**Comment by Rulox** [comment]
Another consideration @Oliphaunte 

If we are going to load the sass from each component, we don't need a `style.scss` file in each folder.

**Comment by Oliphaunte** [comment]
So instead of checking for a main.scss, we look for the scss inside each component?

**Comment by Rulox** [comment]
I think webpack has a way of doing it automatically (ref https://github.com/facebookincubator/create-react-app )

**Comment by Rulox** [comment]
ref. #28

**Comment by Oliphaunte** [comment]
1. Hot reloader is setup
2. Sass is setup

We can remove gulpfile and continue the discussion about using scss in components or main

**Comment by Rulox** [comment]
Bueno.

So, I was thinking about the sass thing. What do you prefer? The main reason to use the "import" way is the reusability of the component across different projects, and to avoid the need of import the style in the main _style.scss file.

What about the approach we have now? Do you see any advantage/disadvantage? @Oliphaunte 

We also need to launch js/sass linters right?

**Comment by Oliphaunte** [comment]
Looking at the js/sass linters issue right now

I think either solution is appropriate, but I agree that the components idea is better for usability. I think for now though, we can leave it as is and add the components change to the issues list, after we iron out all the issues with webpack first

**Comment by Rulox** [comment]
Perfect, makes sense. Then after the lint fix (remember to update travis yml properly), we can merge to develop, and then another PR for the components styles, and if everything works good (that will require some refactor too) we can release a new version

**Comment by Oliphaunte** [comment]
Travis is good now. We have jslint, hot-reload, and sass-compile. 

The last issue is the airbnb linter is breaking on builds. It's working with lint itself though, so not sure if there is a compatibility issue or what. 

Converting all files to jsx fixed the issue with the lint command, but not the builds. I'll see if anything else needs to be done, but if not, we may need another alternative to the airbnb.

**Comment by Rulox** [comment]
Makes sense, thank you @Oliphaunte

File: .eslintrc [hunk 1]
```diff
@@ -19,16 +19,17 @@
...
+    "no-undef": 0, # Should look into this one #
     "eol-last": 0,
     "max-len": ["error", 120],
     "react/forbid-prop-types": 0,
     "object-curly-spacing": 0
   },
   "plugins": [ "react" ],
   "settings": {
-    "import/parser": "babel-eslint",
+    # "import/parser": "babel-eslint", <- Appears to not work in linter
```
Comment (Rulox) [inline .eslintrc]: So, aren't we using this line? can we delete it?

File: package.json [hunk 2]
```diff
@@ -30,29 +29,32 @@
...
+    "eslint-plugin-jsx-a11y": "^2.2.3",
+    "eslint-plugin-react": "^6.6.0",
+    "node-sass": "^3.11.2",
+    "react-hot-loader": "^3.0.0-beta.6",
+    "sass-loader": "^4.0.2",
+    "style-loader": "^0.13.1",
+    "stylelint-webpack-plugin": "^0.4.0",
     "vinyl-buffer": "^1.0.0",
-    "vinyl-source-stream": "^1.1.0"
+    "vinyl-source-stream": "^1.1.0",
```
Comment (Rulox) [inline package.json]: I think we can delete this too

Comment (Oliphaunte) [inline package.json]: What was this for?

**Comment by Oliphaunte** [comment]
Should be good now

**Pull request merged**